
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Ultra-high Sensitivity Mass Spectrometry Single-Cell Proteome Analyses üî¨ üß¨ üíª üß™</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://github.com/Sann5/ultra-prot/README.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="1. Original analyses" href="notebooks/TSP_cell_cycle_analysis.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/logo.png" class="logo" alt="logo">
      
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Notebooks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="notebooks/TSP_cell_cycle_analysis.html">
   1. Original analyses
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="notebooks/documentation.html">
   2. Documentation
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Additional Resources
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference external" href="https://doi.org/10.15252/msb.202110798">
   Ultra-high sensitivity mass spectrometry quantifies single-cell proteome changes upon perturbation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://doi.org/10.1371/journal.pone.0225466">
   HeLa-CCL2 cell heterogeneity studied by single-cell DNA and RNA sequencing. PLoS One 2019
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://doi.org/10.15252/msb.20209946">
   The transcriptome dynamics of single cells during the cell cycle. Mol Syst Biol 2020
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/Sann5/ultra-prot"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/Sann5/ultra-prot/issues/new?title=Issue%20on%20page%20%2FREADME.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/Sann5/ultra-prot/edit/main/README.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Edit this page"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-pencil-alt"></i>
  </span>
<span class="headerbtn__text-container">suggest edit</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="_sources/README.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#why-reproduce-the-analyses">
   Why reproduce the analyses?
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-the-paper-about">
   What is the paper about? üìñ
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#my-two-cents-on-the-paper">
   My two cents on the paper ü™ô ü™ô
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#usage">
   Usage
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#clone-this-repo">
     Clone this repo
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#create-an-environment">
     Create an environment
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#getting-the-data">
     Getting the data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#where-does-the-data-come-from">
     Where does the data come from?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#running-the-analyses">
     Running the analyses
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#running-all-the-analyses">
       Running all the analyses
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#running-a-specific-figure">
       Running a specific figure
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#contributing">
   Contributing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#getting-support">
   Getting support
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#credits">
   Credits
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Ultra-high Sensitivity Mass Spectrometry Single-Cell Proteome Analyses üî¨ üß¨ üíª üß™</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#why-reproduce-the-analyses">
   Why reproduce the analyses?
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-the-paper-about">
   What is the paper about? üìñ
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#my-two-cents-on-the-paper">
   My two cents on the paper ü™ô ü™ô
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#usage">
   Usage
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#clone-this-repo">
     Clone this repo
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#create-an-environment">
     Create an environment
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#getting-the-data">
     Getting the data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#where-does-the-data-come-from">
     Where does the data come from?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#running-the-analyses">
     Running the analyses
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#running-all-the-analyses">
       Running all the analyses
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#running-a-specific-figure">
       Running a specific figure
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#contributing">
   Contributing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#getting-support">
   Getting support
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#credits">
   Credits
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="ultra-high-sensitivity-mass-spectrometry-single-cell-proteome-analyses">
<h1>Ultra-high Sensitivity Mass Spectrometry Single-Cell Proteome Analyses üî¨ üß¨ üíª üß™<a class="headerlink" href="#ultra-high-sensitivity-mass-spectrometry-single-cell-proteome-analyses" title="Permalink to this headline">#</a></h1>
<p>This repository contains a <strong>reproduction</strong> of the analyses made in <a class="reference external" href="https://doi.org/10.15252/msb.202110798">Ultra-high sensitivity mass spectrometry quantifies single-cell proteome changes upon perturbation</a>, A Brunner, ‚Ä¶, M Mann, Molecular Systems Biology (2022), specifically the analyses contained in the associated GitHub <a class="reference external" href="https://github.com/theislab/singlecell_proteomics">repository</a> (excluding the supplementary analyses). You can take a look at a copy of their analyses in our <a class="reference external" href="https://sann5.github.io/ultra-prot/notebooks/TSP_cell_cycle_analysis.html">documentation</a>.</p>
<section id="why-reproduce-the-analyses">
<h2>Why reproduce the analyses?<a class="headerlink" href="#why-reproduce-the-analyses" title="Permalink to this headline">#</a></h2>
<p>We aim to provide the same analyses but in a way that falls in line with various standards and best practices. To restructure the analyses, the reproduction uses <a class="reference external" href="https://snakemake.readthedocs.io/en/stable/">Snakemake</a>, a workflow management system that benefits from improved readability, portability, modularization, transparency, and scalability. Additionally, we tried to improve the repository organization and documentation.</p>
</section>
<section id="what-is-the-paper-about">
<h2>What is the paper about? üìñ<a class="headerlink" href="#what-is-the-paper-about" title="Permalink to this headline">#</a></h2>
<p><em>TLDR: new ultrasensitive single-cell proteomics experimental method.</em></p>
<p>The authors put forward an ultrasensitive mass spectrometry (MS)-based workflow that allows precise and accurate quantification of the proteome of single cells. To keep things short and sweet, here are the major selling points.</p>
<ul class="simple">
<li><p>Ten-fold sensitivity increase.</p></li>
<li><p>Injection of single cells one by one into the MS system allows true single-cell‚Äìderived proteomics.</p></li>
</ul>
<p>This last feature gives a name to their invention; true single-cell‚Äìderived proteomics or T-SCP.</p>
<p>In the second part of the paper -after introducing the technical aspects of their methodology- the authors show that their method is powerful enough to expose differences in the proteome of cells in different stages of the cell cycle. To do so, they took cells from an immortalized human cell line (HeLa cells) and treated different cell populations with different substances that would arrest the cell cycle at different stages. Then they used their method to analyze the proteome of individual cells of the different populations and analyze their differences.</p>
<p>In the third section of the paper, the authors compare the proteomic data with the transcriptomic data of untreated cells of the same lineage. They analyze the expression distributions resulting from both technologies and conclude that the proteome exhibits notably distinct characteristics and therefore cannot entirely be inferred from proteomic data. This idea supports the thesis that although the transcriptome gives way to the proteome, there are a lot of regulatory mechanisms that take place in between.</p>
</section>
<section id="my-two-cents-on-the-paper">
<h2>My two cents on the paper ü™ô ü™ô<a class="headerlink" href="#my-two-cents-on-the-paper" title="Permalink to this headline">#</a></h2>
<p><em>Disclaimer: the following statement reflects my personal opinion and mine only (‚Äúmy‚Äù refers to the author of this repo).</em></p>
<p>The paper spends a significant amount of ‚Äúspace‚Äù demonstrating applications of their method, only to confirm what is currently already thought to be known. I would have spent more effort and content on benchmarking the method.</p>
</section>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">#</a></h2>
<p>You will need <a class="reference external" href="https://docs.conda.io/en/latest/miniconda.html">conda</a>, and <a class="reference external" href="https://git-scm.com/book/en/v2/Getting-Started-Installing-Git">git</a> installed in your computer.</p>
<section id="clone-this-repo">
<h3>Clone this repo<a class="headerlink" href="#clone-this-repo" title="Permalink to this headline">#</a></h3>
<p>If you wish to contribute to this project, fork the repo instead of cloning it.</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>git<span class="w"> </span>clone<span class="w"> </span>git@github.com:Sann5/ultra-prot.git
</pre></div>
</div>
</section>
<section id="create-an-environment">
<h3>Create an environment<a class="headerlink" href="#create-an-environment" title="Permalink to this headline">#</a></h3>
<p>Using a virtual environment is strongly recommended. If you use <code class="docutils literal notranslate"><span class="pre">conda</span></code> you can create an environment with the following command. Before you do place yourself at the root of the repo.</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>ultra-prot
</pre></div>
</div>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-n<span class="w"> </span>ultra-prot<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>.9<span class="w"> </span>-f<span class="w"> </span>environment.yml
</pre></div>
</div>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>activate<span class="w"> </span>ultra-prot
</pre></div>
</div>
</section>
<section id="getting-the-data">
<h3>Getting the data<a class="headerlink" href="#getting-the-data" title="Permalink to this headline">#</a></h3>
<p>Download the raw data by running the following command. Be warned, doing so can take some time (around 20 minutes). Before you do so, make sure you are positioned at the root directory.</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span><span class="nb">pwd</span>
&lt;location_of_your_clone&gt;/ultra-prot
</pre></div>
</div>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>./src/download_raw.sh
</pre></div>
</div>
<p>Alternatively, if you would like to download the proteomic and the transcriptomic data separately you can run the following commands.</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>./src/download_proteomic.sh
</pre></div>
</div>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>./src/download_transcriptomic.sh
</pre></div>
</div>
</section>
<section id="where-does-the-data-come-from">
<h3>Where does the data come from?<a class="headerlink" href="#where-does-the-data-come-from" title="Permalink to this headline">#</a></h3>
<p>For the analyses the authors used:</p>
<ul class="simple">
<li><p>Data generated by the experiments they present in the paper.</p>
<ul>
<li><p><a class="reference external" href="https://www.ebi.ac.uk/pride/archive/projects/PXD024043">Web page of raw data</a> (file name: <code class="docutils literal notranslate"><span class="pre">DIANN1.8_SingleCells_CellCycle.7z</span></code>)</p></li>
<li><p>Once you download the raw data you can find it in <code class="docutils literal notranslate"><span class="pre">data/raw/proteomic</span></code>.</p></li>
<li><p>Once you run the workflow you can find the preprocessed data in <code class="docutils literal notranslate"><span class="pre">/data/preprocessed/proteomic</span></code>.</p></li>
</ul>
</li>
<li><p>Annotation data that is provided with the original <a class="reference external" href="https://github.com/theislab/singlecell_proteomics">repository</a> (file names: <code class="docutils literal notranslate"><span class="pre">CellCycleMarker.xlsx</span></code> and <code class="docutils literal notranslate"><span class="pre">CoreProteome.txt</span></code>). Here you will find it in this <a class="reference external" href="https://github.com/Sann5/ultra-prot/tree/main/data/preprocessed/other">folder</a>.</p></li>
<li><p>Transcriptomic data from two other studies.</p>
<ul>
<li><p><a class="reference external" href="https://doi.org/10.1371/journal.pone.0225466">HeLa-CCL2 cell heterogeneity studied by single-cell DNA and RNA sequencing. PLoS One 2019</a>.</p>
<ul>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129447">Web page of raw dataset</a>.</p></li>
<li><p>Once downloaded you can find the raw data in <code class="docutils literal notranslate"><span class="pre">data/raw/transcriptomic/study_1</span></code>.</p></li>
<li><p>Once you run the workflow you can find the preprocessed data in <code class="docutils literal notranslate"><span class="pre">data/preprocessed/transcriptomic/GSM3713084_HeLa.h5ad</span></code>.</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://doi.org/10.15252/msb.20209946">The transcriptome dynamics of single cells during the cell cycle. Mol Syst Biol 2020</a>.</p>
<ul>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE142277">Web page of raw dataset 1</a>.</p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4226257">Web page of raw dataset 2</a>.</p></li>
<li><p>Once downloaded you can find the raw data in <code class="docutils literal notranslate"><span class="pre">data/raw/transcriptomic/study_2</span></code>.</p></li>
<li><p>Once you run the workflow you can find the preprocessed data in <code class="docutils literal notranslate"><span class="pre">data/preprocessed/transcriptomic/GSM4226257_out_gene_exon_tagged.h5ad</span></code>.</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
<section id="running-the-analyses">
<h3>Running the analyses<a class="headerlink" href="#running-the-analyses" title="Permalink to this headline">#</a></h3>
<p>Thanks to Snakemake, running the entire workflow or individual parts is easy. Each figure in the paper represents one single analysis. The workflow generates each of the figures and saves them to the results folder (<code class="docutils literal notranslate"><span class="pre">./results</span></code> which is created by the workflow). All the preprocessing will be managed by Snakemake, so there is no need to explicitly run it. Just need to make sure the raw data is there before running it.</p>
<section id="running-all-the-analyses">
<h4>Running all the analyses<a class="headerlink" href="#running-all-the-analyses" title="Permalink to this headline">#</a></h4>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>snakemake<span class="w"> </span>--cores<span class="w"> </span><span class="m">1</span>
</pre></div>
</div>
<p><em>You can always adjust the number of cores according to your computational resources.</em></p>
</section>
<section id="running-a-specific-figure">
<h4>Running a specific figure<a class="headerlink" href="#running-a-specific-figure" title="Permalink to this headline">#</a></h4>
<p>The following figures are considered and therefore valid for input: <code class="docutils literal notranslate"><span class="pre">fig_4c</span> <span class="pre">fig_4d</span> <span class="pre">fig_4e</span> <span class="pre">fig_5c</span> <span class="pre">fig_5d</span> <span class="pre">fig_5e</span> <span class="pre">fig_5a</span> <span class="pre">fig_5b</span></code>. You can check the source code for each figure in <code class="docutils literal notranslate"><span class="pre">src/fig_*.py</span></code>. For example, if we wanted to run generate only figure 4d we would run the following command.</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>snakemake<span class="w"> </span>--cores<span class="w"> </span><span class="m">1</span><span class="w"> </span>fig_4d
</pre></div>
</div>
</section>
</section>
</section>
<section id="contributing">
<h2>Contributing<a class="headerlink" href="#contributing" title="Permalink to this headline">#</a></h2>
<p>Check <a class="reference external" href="https://github.com/Sann5/ultra-prot/blob/main/CONTRIBUTING.md">CONTRIBUTING.md</a> for our contributing guidelines and values. If you would like to contribute but don‚Äôt know exactly what, here are a few ideas:</p>
<ul class="contains-task-list simple">
<li class="task-list-item"><p><input class="task-list-item-checkbox" disabled="disabled" type="checkbox"> Parametrize the workflow with <a class="reference external" href="https://snakemake.readthedocs.io/en/stable/tutorial/advanced.html#step-2-config-files">Snakemake‚Äôs config file mechanism</a>.</p></li>
<li class="task-list-item"><p><input class="task-list-item-checkbox" disabled="disabled" type="checkbox"> Make appropriate documentation for each of the analyses entailed by each of the figures (<a class="reference external" href="https://sann5.github.io/ultra-prot/notebooks/documentation.html">this documentation is just an example</a>).</p></li>
<li class="task-list-item"><p><input class="task-list-item-checkbox" disabled="disabled" type="checkbox"> Extend the workflow such that it also includes the figures in the supplementary material of the paper.</p></li>
<li class="task-list-item"><p><input class="task-list-item-checkbox" disabled="disabled" type="checkbox"> Add dome unit and integration <a class="reference external" href="https://github.com/Sann5/ultra-prot/tree/main/tests">tests</a>.</p></li>
</ul>
</section>
<section id="getting-support">
<h2>Getting support<a class="headerlink" href="#getting-support" title="Permalink to this headline">#</a></h2>
<p>Check <a class="reference external" href="https://github.com/Sann5/ultra-prot/blob/main/SUPPORT.md">SUPPORT.md</a>.</p>
</section>
<section id="credits">
<h2>Credits<a class="headerlink" href="#credits" title="Permalink to this headline">#</a></h2>
<ul class="simple">
<li><p>This project was created using a template from the <a class="reference external" href="https://worldbank.github.io/template/README.html">World Bank Development Data Group</a>.</p></li>
<li><p>All the analyses are almost exact copies of the code in <a class="reference external" href="https://github.com/theislab/singlecell_proteomics/blob/main/TSP_cell_cycle_analysis.ipynb">this notebook</a>.</p></li>
<li><p>Logo downloaded from <a class="reference external" href="https://favpng.com/png_view/cells-icon-single-cell-sequencing-clip-art-mitosis-cytokinesis-cell-division-png/jg6yPfYZ">here</a>.</p></li>
</ul>
</section>
<div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='right-next' id="next-link" href="notebooks/TSP_cell_cycle_analysis.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">1. </span>Original analyses</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Santiago Castro Dau<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>