# Ultra-high Sensitivity Mass Spectrometry Single-Cell Proteome Analyses ðŸ”¬ ðŸ§¬ ðŸ’» ðŸ§ª

This repository contains a **reproduction** of the analyses made in [Ultra-high sensitivity mass spectrometry quantifies single-cell proteome changes upon perturbation](https://doi.org/10.15252/msb.202110798), A Brunner, ..., M Mann, Molecular Systems Biology (2022), specifically the analyses contained in the associated GitHub [repository](https://github.com/theislab/singlecell_proteomics) (excluding the supplementary analyses).

## Why reproduce the analyses?

We aim to provide the same analyses but in a way that falls in line with various standards and best practices. To restructure the analyses, the reproduction uses [Snakemake](https://snakemake.readthedocs.io/en/stable/), a workflow management system that benefits from improved readability, portability, modularization, transparency, and scalability. Additionally, we tried to improve the repository organization and documentation.

## What is the paper about? ðŸ“–

*TLDR: new ultrasensitive single-cell proteomics experimental method.*

The authors put forward an ultrasensitive mass spectrometry (MS)-based workflow that allows precise and accurate quantification of the proteome of single cells. To keep things short and sweet, here are the major selling points.

- Ten-fold sensitivity increase.
- Injection of single cells one by one into the MS system allows true single-cellâ€“derived proteomics.

This last feature gives a name to their invention; true single-cellâ€“derived proteomics or T-SCP.

In the second part of the paper -after introducing the technical aspects of their methodology- the authors show that their method is powerful enough to expose differences in the proteome of cells in different stages of the cell cycle. To do so, they took cells from an immortalized human cell line (HeLa cells) and treated different cell populations with different substances that would arrest the cell cycle at different stages. Then they used their method to analyze the proteome of individual cells of the different populations and analyze their differences.

In the third section of the paper, the authors compare the proteomic data with the transcriptomic data of untreated cells of the same lineage. They analyze the expression distributions resulting from both technologies and conclude that the proteome exhibits notably distinct characteristics and therefore cannot entirely be inferred from proteomic data. This idea supports the thesis that although the transcriptome gives way to the proteome, there are a lot of regulatory mechanisms that take place in between.

## My two cents on the paper ðŸª™ ðŸª™

*Disclaimer: the following statement reflects my personal opinion and mine only ("my" refers to the author of this repo).*

The paper spends a significant amount of "space" demonstrating applications of their method, only to confirm what is currently already thought to be known. I would have spent more effort and content on benchmarking the method.

## Usage

### Clone this repo

If you wish to contribute to this project, fork the repo instead of cloning it.

```sh
git clone git@github.com:Sann5/ultra-prot.git
```

### Create an environment

Using a virtual environment is strongly recommended. If you use `conda` you can create an environment with the following command.

```sh
conda env create -n ultra -f environment.yml
```

### Getting the data

Download the raw data by running the following command. Be warned, doing so can take some time (around 20 minutes). Before you do so, make sure you are positioned at the root directory.

```sh
$ pwd
<location_of_your_clone>/ultra-prot
```

```sh
./src/download_raw.sh
```

Alternatively, if you would like to download the proteomic and the transcriptomic data separately you can run the following commands.

```sh
./src/download_proteomic.sh
```

```sh
./src/download_transcriptomic.sh
```

### Where does the data come from?

For the analyses the authors used:

- Data generated by the experiments they present in the paper.
  - [Web page of raw data](https://www.ebi.ac.uk/pride/archive/projects/PXD024043) (file name: `DIANN1.8_SingleCells_CellCycle.7z`)
  - Once you download the raw data you can find it in this [folder](.github/data/raw/proteomic).
  - Once you run the workflow you can find the preprocessed data in this [folder](.github/data/preprocessed/proteomic).
- Annotation data that is provided with the original [repository](https://github.com/theislab/singlecell_proteomics) (file names: `CellCycleMarker.xlsx` and `CoreProteome.txt`). Here you will find it in this [folder](.github/data/raw/other).
- Transcriptomic data from two other studies.
  - [HeLa-CCL2 cell heterogeneity studied by single-cell DNA and RNA sequencing. PLoS One 2019](https://doi.org/10.1371/journal.pone.0225466)
    - [Web page of raw data](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129447)
    - Once downloaded you can find the raw data in this [folder](data/raw/transcriptomic/study_1).
    - Once you run the workflow you can find the preprocessed data in this [folder](.github/data/preprocessed/transcriptomic) (file name: `GSM3713084_HeLa.h5ad`)
  - [The transcriptome dynamics of single cells during the cell cycle. Mol Syst Biol 2020](https://doi.org/10.15252/msb.20209946)
    - [Web page of raw dataset 1](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE142277).
    - [Web page of raw dataset 2](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4226257).
    - Once downloaded you can find the raw data in this [folder](data/raw/transcriptomic/study_2) (file name: `GSM4226257_out_gene_exon_tagged.h5ad`).

### Running the analyses

Thanks to Snakemake, running the entire workflow or individual parts is easy. Each figure in the paper represents one single analysis. The workflow generates each of the figures and saves them to the [results folder](.github/results). All the preprocessing will be managed by Snakemake, so there is no need to explicitly run it. Just need to make sure the raw data is there before running it.

#### Running all the analyses

```sh
snakemake --cores 1
```

*You can always adjust the number of cores according to your computational resources.*

#### Running a specific figure

The following figures are considered and therefore valid for input: `fig_4c fig_4d fig_4e fig_5c fig_5d fig_5e fig_5a fig_5b`. You can check the source code for each figure in `src/fig_*.py`. For example, if we wanted to run generate only figure 4d we would run the following command.

```sh
snakemake --cores 1 fig_4d
```

## Contributing

Check [CONTRIBUTING.md](.github/CONTRIBUTING.md) for our contributing guidelines and values. If you would like to contribute but don't know exactly what, here are a few ideas:

- [ ] Parametrize the workflow with [Snakemake's config file mechanism](https://snakemake.readthedocs.io/en/stable/tutorial/advanced.html#step-2-config-files).
- [ ] Make appropriate documentation for each of the analyses entailed by each of the figures ([this documentation is just an example](.github/notebooks/documentation.ipynb)).
- [ ] Extend the workflow such that it also includes the figures in the supplementary material of the paper.
- [ ] Add dome unit and integration [tests](.github/tests).

## Getting support

Check [SUPPORT.md](.github/SUPPORT.md).

## Credits

This project was created using a template from the [World Bank Development Data Group](https://worldbank.github.io/template/README.html). All the analyses are almost exact copies of the code in [this notebook](https://github.com/theislab/singlecell_proteomics/blob/main/TSP_cell_cycle_analysis.ipynb).
