
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>1. Data Loading</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://github.com/Sann5/ultra-prot/notebooks/TSP_cell_cycle_analysis.html" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2. Documentation" href="documentation.html" />
    <link rel="prev" title="Ultra-high Sensitivity Mass Spectrometry Single-Cell Proteome Analysis ðŸ”¬ ðŸ§¬ ðŸ’» ðŸ§ª" href="../README.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Examples
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   1. Data Loading
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="documentation.html">
   2. Documentation
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Additional Resources
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference external" href="https://doi.org/10.15252/msb.202110798">
   Ultra-high sensitivity mass spectrometry quantifies single-cell proteome changes upon perturbation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://doi.org/10.1371/journal.pone.0225466">
   HeLa-CCL2 cell heterogeneity studied by single-cell DNA and RNA sequencing. PLoS One 2019
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://doi.org/10.15252/msb.20209946">
   The transcriptome dynamics of single cells during the cell cycle. Mol Syst Biol 2020
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/Sann5/ultra-prot/main?urlpath=tree/notebooks/TSP_cell_cycle_analysis.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="../_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/Sann5/ultra-prot"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/Sann5/ultra-prot/issues/new?title=Issue%20on%20page%20%2Fnotebooks/TSP_cell_cycle_analysis.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/Sann5/ultra-prot/edit/main/notebooks/TSP_cell_cycle_analysis.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Edit this page"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-pencil-alt"></i>
  </span>
<span class="headerbtn__text-container">suggest edit</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/notebooks/TSP_cell_cycle_analysis.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#true-single-cell-proteomics-cell-cycle-analysis">
   1.1. True Single-Cell Proteomics - Cell Cycle Analysis
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-4c-total-protein-signal-per-cell-grouped-by-cell-cycle-stage">
     1.1.1. Fig. 4C - Total protein signal per cell grouped by cell cycle stage
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-4d-principal-component-analysis">
     1.1.2. Fig. 4D - Principal Component Analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-4e-cell-cycle-stage-prediction">
     1.1.3. Fig. 4E - Cell cycle stage prediction
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#true-single-cell-proteomics-comparison-with-transcriptomics-data">
   1.2. True Single-Cell Proteomics - Comparison with Transcriptomics Data
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5a-gene-expression-completeness-within-cells">
     1.2.1. Fig. 5A - Gene expression completeness within cells
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5b-principal-component-analysis">
     1.2.2. Fig. 5B - Principal Component Analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5c-pearson-correlation-analysis">
     1.2.3. Fig. 5C - Pearson Correlation Analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5d-variance-analysis">
     1.2.4. Fig. 5D - Variance analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5e-variance-comparison">
     1.2.5. Fig. 5E - Variance comparison
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#supplementary-analyses">
   1.3. Supplementary Analyses
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#effect-of-local-regression-normalisation">
     1.3.1. Effect of Local Regression Normalisation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#pca-including-untreated-cells">
     1.3.2. PCA including untreated cells
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cell-cycle-stage-prediction">
     1.3.3. Cell cycle stage prediction
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-completeness">
     1.3.4. Data completeness
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#variance-analyses">
     1.3.5. Variance analyses
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#correlation-analyses">
     1.3.6. Correlation analyses
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#abundance-value-distributions">
     1.3.7. Abundance value distributions
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-before-and-after-filtering">
     1.3.8. Data before and after filtering
    </a>
   </li>
  </ul>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Data Loading</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#true-single-cell-proteomics-cell-cycle-analysis">
   1.1. True Single-Cell Proteomics - Cell Cycle Analysis
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-4c-total-protein-signal-per-cell-grouped-by-cell-cycle-stage">
     1.1.1. Fig. 4C - Total protein signal per cell grouped by cell cycle stage
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-4d-principal-component-analysis">
     1.1.2. Fig. 4D - Principal Component Analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-4e-cell-cycle-stage-prediction">
     1.1.3. Fig. 4E - Cell cycle stage prediction
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#true-single-cell-proteomics-comparison-with-transcriptomics-data">
   1.2. True Single-Cell Proteomics - Comparison with Transcriptomics Data
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5a-gene-expression-completeness-within-cells">
     1.2.1. Fig. 5A - Gene expression completeness within cells
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5b-principal-component-analysis">
     1.2.2. Fig. 5B - Principal Component Analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5c-pearson-correlation-analysis">
     1.2.3. Fig. 5C - Pearson Correlation Analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5d-variance-analysis">
     1.2.4. Fig. 5D - Variance analysis
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fig-5e-variance-comparison">
     1.2.5. Fig. 5E - Variance comparison
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#supplementary-analyses">
   1.3. Supplementary Analyses
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#effect-of-local-regression-normalisation">
     1.3.1. Effect of Local Regression Normalisation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#pca-including-untreated-cells">
     1.3.2. PCA including untreated cells
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cell-cycle-stage-prediction">
     1.3.3. Cell cycle stage prediction
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-completeness">
     1.3.4. Data completeness
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#variance-analyses">
     1.3.5. Variance analyses
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#correlation-analyses">
     1.3.6. Correlation analyses
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#abundance-value-distributions">
     1.3.7. Abundance value distributions
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-before-and-after-filtering">
     1.3.8. Data before and after filtering
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">anndata</span>
<span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">matplotlib.cm</span> <span class="k">as</span> <span class="nn">cm</span>
<span class="kn">from</span> <span class="nn">matplotlib.colors</span> <span class="kn">import</span> <span class="n">ListedColormap</span>
<span class="kn">import</span> <span class="nn">scipy</span> 
<span class="kn">import</span> <span class="nn">sklearn.metrics</span>

<span class="n">plt</span><span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">use</span><span class="p">(</span><span class="s1">&#39;seaborn-colorblind&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">rc</span><span class="p">(</span><span class="s1">&#39;xtick&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">11</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">rc</span><span class="p">(</span><span class="s1">&#39;ytick&#39;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">11</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s1">&#39;font.size&#39;</span><span class="p">:</span> <span class="mi">11</span><span class="p">})</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/tmp/ipykernel_7950/4079999522.py:12: MatplotlibDeprecationWarning: The seaborn styles shipped by Matplotlib are deprecated since 3.6, as they no longer correspond to the styles shipped by seaborn. However, they will remain available as &#39;seaborn-v0_8-&lt;style&gt;&#39;. Alternatively, directly use the seaborn API instead.
  plt.style.use(&#39;seaborn-colorblind&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">platform</span> <span class="kn">import</span> <span class="n">python_version</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Python version: </span><span class="si">{</span><span class="n">python_version</span><span class="p">()</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Scanpy version: </span><span class="si">{</span><span class="n">sc</span><span class="o">.</span><span class="n">__version__</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Python version: 3.9.15
Scanpy version: 1.9.1
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data_path</span> <span class="o">=</span> <span class="s1">&#39;data/&#39;</span>
</pre></div>
</div>
</div>
</div>
<section class="tex2jax_ignore mathjax_ignore" id="data-loading">
<h1><span class="section-number">1. </span>Data Loading<a class="headerlink" href="#data-loading" title="Permalink to this headline">#</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># load raw protein dataset</span>

<span class="n">filename</span> <span class="o">=</span> <span class="s1">&#39;20210919_DIANN_SingleCellOutput.pg_matrix_notnormalized.tsv&#39;</span>

<span class="n">cols</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">))</span>
<span class="n">prot</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">usecols</span> <span class="o">=</span><span class="p">[</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">cols</span> <span class="k">if</span> <span class="s1">&#39;TSCP_DIA_SingleCell&#39;</span> <span class="ow">in</span> <span class="n">i</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">var_annotation</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">index_col</span> <span class="o">=</span> <span class="s1">&#39;Genes&#39;</span><span class="p">,</span> <span class="n">usecols</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Genes&#39;</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">prot</span> <span class="o">=</span> <span class="n">prot</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mf">0.</span><span class="p">)</span>
<span class="n">prot</span><span class="p">[</span><span class="n">prot</span> <span class="o">==</span> <span class="s1">&#39;Filtered&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mf">0.</span>

<span class="n">adata_prot_raw</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">prot</span><span class="o">.</span><span class="n">transpose</span><span class="p">())</span>
<span class="n">adata_prot_raw</span><span class="o">.</span><span class="n">var</span> <span class="o">=</span> <span class="n">var_annotation</span>
<span class="n">adata_prot_raw</span><span class="o">.</span><span class="n">var_names</span> <span class="o">=</span> <span class="n">adata_prot_raw</span><span class="o">.</span><span class="n">var_names</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span>
<span class="n">adata_prot_raw</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">FileNotFoundError</span><span class="g g-Whitespace">                         </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">4</span><span class="p">],</span> <span class="n">line</span> <span class="mi">5</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># load raw protein dataset</span>
<span class="g g-Whitespace">      </span><span class="mi">3</span> <span class="n">filename</span> <span class="o">=</span> <span class="s1">&#39;20210919_DIANN_SingleCellOutput.pg_matrix_notnormalized.tsv&#39;</span>
<span class="ne">----&gt; </span><span class="mi">5</span> <span class="n">cols</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">))</span>
<span class="g g-Whitespace">      </span><span class="mi">6</span> <span class="n">prot</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">usecols</span> <span class="o">=</span><span class="p">[</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">cols</span> <span class="k">if</span> <span class="s1">&#39;TSCP_DIA_SingleCell&#39;</span> <span class="ow">in</span> <span class="n">i</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">7</span> <span class="n">var_annotation</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">index_col</span> <span class="o">=</span> <span class="s1">&#39;Genes&#39;</span><span class="p">,</span> <span class="n">usecols</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Genes&#39;</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="nn">File /usr/share/miniconda/envs/test/lib/python3.9/site-packages/pandas/util/_decorators.py:311,</span> in <span class="ni">deprecate_nonkeyword_arguments.&lt;locals&gt;.decorate.&lt;locals&gt;.wrapper</span><span class="nt">(*args, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">305</span> <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">args</span><span class="p">)</span> <span class="o">&gt;</span> <span class="n">num_allow_args</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">306</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">307</span>         <span class="n">msg</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">arguments</span><span class="o">=</span><span class="n">arguments</span><span class="p">),</span>
<span class="g g-Whitespace">    </span><span class="mi">308</span>         <span class="ne">FutureWarning</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">309</span>         <span class="n">stacklevel</span><span class="o">=</span><span class="n">stacklevel</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">310</span>     <span class="p">)</span>
<span class="ne">--&gt; </span><span class="mi">311</span> <span class="k">return</span> <span class="n">func</span><span class="p">(</span><span class="o">*</span><span class="n">args</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>

<span class="nn">File /usr/share/miniconda/envs/test/lib/python3.9/site-packages/pandas/io/parsers/readers.py:680,</span> in <span class="ni">read_csv</span><span class="nt">(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)</span>
<span class="g g-Whitespace">    </span><span class="mi">665</span> <span class="n">kwds_defaults</span> <span class="o">=</span> <span class="n">_refine_defaults_read</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">666</span>     <span class="n">dialect</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">667</span>     <span class="n">delimiter</span><span class="p">,</span>
   <span class="p">(</span><span class="o">...</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">676</span>     <span class="n">defaults</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;delimiter&quot;</span><span class="p">:</span> <span class="s2">&quot;,&quot;</span><span class="p">},</span>
<span class="g g-Whitespace">    </span><span class="mi">677</span> <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">678</span> <span class="n">kwds</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">kwds_defaults</span><span class="p">)</span>
<span class="ne">--&gt; </span><span class="mi">680</span> <span class="k">return</span> <span class="n">_read</span><span class="p">(</span><span class="n">filepath_or_buffer</span><span class="p">,</span> <span class="n">kwds</span><span class="p">)</span>

<span class="nn">File /usr/share/miniconda/envs/test/lib/python3.9/site-packages/pandas/io/parsers/readers.py:575,</span> in <span class="ni">_read</span><span class="nt">(filepath_or_buffer, kwds)</span>
<span class="g g-Whitespace">    </span><span class="mi">572</span> <span class="n">_validate_names</span><span class="p">(</span><span class="n">kwds</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;names&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">))</span>
<span class="g g-Whitespace">    </span><span class="mi">574</span> <span class="c1"># Create the parser.</span>
<span class="ne">--&gt; </span><span class="mi">575</span> <span class="n">parser</span> <span class="o">=</span> <span class="n">TextFileReader</span><span class="p">(</span><span class="n">filepath_or_buffer</span><span class="p">,</span> <span class="o">**</span><span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">577</span> <span class="k">if</span> <span class="n">chunksize</span> <span class="ow">or</span> <span class="n">iterator</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">578</span>     <span class="k">return</span> <span class="n">parser</span>

<span class="nn">File /usr/share/miniconda/envs/test/lib/python3.9/site-packages/pandas/io/parsers/readers.py:934,</span> in <span class="ni">TextFileReader.__init__</span><span class="nt">(self, f, engine, **kwds)</span>
<span class="g g-Whitespace">    </span><span class="mi">931</span>     <span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="p">[</span><span class="s2">&quot;has_index_names&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">kwds</span><span class="p">[</span><span class="s2">&quot;has_index_names&quot;</span><span class="p">]</span>
<span class="g g-Whitespace">    </span><span class="mi">933</span> <span class="bp">self</span><span class="o">.</span><span class="n">handles</span><span class="p">:</span> <span class="n">IOHandles</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
<span class="ne">--&gt; </span><span class="mi">934</span> <span class="bp">self</span><span class="o">.</span><span class="n">_engine</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_make_engine</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">engine</span><span class="p">)</span>

<span class="nn">File /usr/share/miniconda/envs/test/lib/python3.9/site-packages/pandas/io/parsers/readers.py:1218,</span> in <span class="ni">TextFileReader._make_engine</span><span class="nt">(self, f, engine)</span>
<span class="g g-Whitespace">   </span><span class="mi">1214</span>     <span class="n">mode</span> <span class="o">=</span> <span class="s2">&quot;rb&quot;</span>
<span class="g g-Whitespace">   </span><span class="mi">1215</span> <span class="c1"># error: No overload variant of &quot;get_handle&quot; matches argument types</span>
<span class="g g-Whitespace">   </span><span class="mi">1216</span> <span class="c1"># &quot;Union[str, PathLike[str], ReadCsvBuffer[bytes], ReadCsvBuffer[str]]&quot;</span>
<span class="g g-Whitespace">   </span><span class="mi">1217</span> <span class="c1"># , &quot;str&quot;, &quot;bool&quot;, &quot;Any&quot;, &quot;Any&quot;, &quot;Any&quot;, &quot;Any&quot;, &quot;Any&quot;</span>
<span class="ne">-&gt; </span><span class="mi">1218</span> <span class="bp">self</span><span class="o">.</span><span class="n">handles</span> <span class="o">=</span> <span class="n">get_handle</span><span class="p">(</span>  <span class="c1"># type: ignore[call-overload]</span>
<span class="g g-Whitespace">   </span><span class="mi">1219</span>     <span class="n">f</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1220</span>     <span class="n">mode</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1221</span>     <span class="n">encoding</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;encoding&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">compression</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;compression&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="n">memory_map</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;memory_map&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1224</span>     <span class="n">is_text</span><span class="o">=</span><span class="n">is_text</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1225</span>     <span class="n">errors</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;encoding_errors&quot;</span><span class="p">,</span> <span class="s2">&quot;strict&quot;</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">storage_options</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;storage_options&quot;</span><span class="p">,</span> <span class="kc">None</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span> <span class="k">assert</span> <span class="bp">self</span><span class="o">.</span><span class="n">handles</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span> <span class="n">f</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">handles</span><span class="o">.</span><span class="n">handle</span>

<span class="nn">File /usr/share/miniconda/envs/test/lib/python3.9/site-packages/pandas/io/common.py:786,</span> in <span class="ni">get_handle</span><span class="nt">(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)</span>
<span class="g g-Whitespace">    </span><span class="mi">781</span> <span class="k">elif</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">handle</span><span class="p">,</span> <span class="nb">str</span><span class="p">):</span>
<span class="g g-Whitespace">    </span><span class="mi">782</span>     <span class="c1"># Check whether the filename is to be opened in binary mode.</span>
<span class="g g-Whitespace">    </span><span class="mi">783</span>     <span class="c1"># Binary mode does not support &#39;encoding&#39; and &#39;newline&#39;.</span>
<span class="g g-Whitespace">    </span><span class="mi">784</span>     <span class="k">if</span> <span class="n">ioargs</span><span class="o">.</span><span class="n">encoding</span> <span class="ow">and</span> <span class="s2">&quot;b&quot;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">ioargs</span><span class="o">.</span><span class="n">mode</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">785</span>         <span class="c1"># Encoding</span>
<span class="ne">--&gt; </span><span class="mi">786</span>         <span class="n">handle</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">787</span>             <span class="n">handle</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">788</span>             <span class="n">ioargs</span><span class="o">.</span><span class="n">mode</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">789</span>             <span class="n">encoding</span><span class="o">=</span><span class="n">ioargs</span><span class="o">.</span><span class="n">encoding</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">790</span>             <span class="n">errors</span><span class="o">=</span><span class="n">errors</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">791</span>             <span class="n">newline</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">    </span><span class="mi">792</span>         <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">793</span>     <span class="k">else</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">794</span>         <span class="c1"># Binary mode</span>
<span class="g g-Whitespace">    </span><span class="mi">795</span>         <span class="n">handle</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">handle</span><span class="p">,</span> <span class="n">ioargs</span><span class="o">.</span><span class="n">mode</span><span class="p">)</span>

<span class="ne">FileNotFoundError</span>: [Errno 2] No such file or directory: &#39;data/20210919_DIANN_SingleCellOutput.pg_matrix_notnormalized.tsv&#39;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># load protein dataset with local regression normalization</span>

<span class="n">filename</span> <span class="o">=</span> <span class="s1">&#39;20210919_DIANN_SingleCellOutput.pg_matrix_cellcyclepred.tsv&#39;</span>

<span class="n">cols</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">))</span>
<span class="n">prot</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">usecols</span> <span class="o">=</span><span class="p">[</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">cols</span> <span class="k">if</span> <span class="s1">&#39;TSCP_DIA_SingleCell&#39;</span> <span class="ow">in</span> <span class="n">i</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">var_annotation</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="n">filename</span><span class="p">,</span> <span class="n">index_col</span> <span class="o">=</span> <span class="s1">&#39;Genes&#39;</span><span class="p">,</span> <span class="n">usecols</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Genes&#39;</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">prot</span><span class="p">[</span><span class="n">prot</span> <span class="o">==</span> <span class="s1">&#39;Filtered&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>

<span class="n">adata_prot</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">AnnData</span><span class="p">(</span><span class="n">prot</span><span class="o">.</span><span class="n">transpose</span><span class="p">())</span>
<span class="n">adata_prot</span><span class="o">.</span><span class="n">var</span> <span class="o">=</span> <span class="n">var_annotation</span>
<span class="n">adata_prot</span><span class="o">.</span><span class="n">var_names</span> <span class="o">=</span> <span class="n">adata_prot</span><span class="o">.</span><span class="n">var_names</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span>
<span class="n">adata_prot</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>

<span class="n">protein_adata</span> <span class="o">=</span> <span class="n">adata_prot</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># load data from &quot;HeLa-CCL2 cell heterogeneity studied by single-cell DNA and RNA sequencing&quot;</span>
<span class="c1"># https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129447</span>
<span class="c1"># (SMART-Seq2 data from 9, 14 and 20 passages)</span>

<span class="n">adata1</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE129447_RAW/GSM3713084_HeLa_1.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata2</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE129447_RAW/GSM3713085_HeLa_2.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata3</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE129447_RAW/GSM3713086_HeLa_3.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata3</span><span class="o">.</span><span class="n">var_names</span> <span class="o">=</span> <span class="n">adata1</span><span class="o">.</span><span class="n">var_names</span>
<span class="n">adata2</span> <span class="o">=</span> <span class="n">adata2</span><span class="p">[:,</span> <span class="n">adata1</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span> <span class="c1"># without ERCC genes</span>
<span class="n">smartseq_adata</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">adata1</span><span class="p">,</span> <span class="n">adata2</span><span class="p">,</span> <span class="n">adata3</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Variable names are not unique. To make them unique, call `.var_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># load data from &quot;The transcriptome dynamics of single cells during the cell cycle&quot;</span>
<span class="c1"># https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE142277</span>

<span class="n">adata1_exon</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE142277_RAW/GSM4224315_out_gene_exon_tagged.dge_exonssf002_WT.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata1_intron</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE142277_RAW/GSM4224315_out_gene_exon_tagged.dge_intronssf002_WT.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata2_exon</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE142277_RAW/GSM4224316_out_gene_exon_tagged.dge_exonssf002_KO.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata2_intron</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSE142277_RAW/GSM4224316_out_gene_exon_tagged.dge_intronssf002_KO.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>

<span class="n">intersection</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">adata1_exon</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata1_intron</span><span class="o">.</span><span class="n">var_names</span><span class="p">))</span>
<span class="n">intersection_part</span> <span class="o">=</span> <span class="n">adata1_exon</span><span class="p">[:,</span> <span class="n">intersection</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">intersection_part</span><span class="o">.</span><span class="n">X</span> <span class="o">+=</span> <span class="n">adata1_intron</span><span class="p">[:,</span> <span class="n">intersection</span><span class="p">]</span><span class="o">.</span><span class="n">X</span>
<span class="n">adata1_2</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span>
        <span class="n">intersection_part</span><span class="p">,</span> 
        <span class="n">adata1_exon</span><span class="p">[:,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">adata1_exon</span><span class="o">.</span><span class="n">var_names</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata1_intron</span><span class="o">.</span><span class="n">var_names</span><span class="p">]],</span>
        <span class="n">adata1_intron</span><span class="p">[:,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">adata1_intron</span><span class="o">.</span><span class="n">var_names</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata1_exon</span><span class="o">.</span><span class="n">var_names</span><span class="p">]],</span>
    <span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="n">intersection</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">adata2_exon</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata2_intron</span><span class="o">.</span><span class="n">var_names</span><span class="p">))</span>
<span class="n">intersection_part</span> <span class="o">=</span> <span class="n">adata2_exon</span><span class="p">[:,</span> <span class="n">intersection</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">intersection_part</span><span class="o">.</span><span class="n">X</span> <span class="o">+=</span> <span class="n">adata2_intron</span><span class="p">[:,</span> <span class="n">intersection</span><span class="p">]</span><span class="o">.</span><span class="n">X</span>
<span class="n">adata2_2</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span>
        <span class="n">intersection_part</span><span class="p">,</span> 
        <span class="n">adata2_exon</span><span class="p">[:,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">adata2_exon</span><span class="o">.</span><span class="n">var_names</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata2_intron</span><span class="o">.</span><span class="n">var_names</span><span class="p">]],</span>
        <span class="n">adata2_intron</span><span class="p">[:,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">adata2_intron</span><span class="o">.</span><span class="n">var_names</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata2_exon</span><span class="o">.</span><span class="n">var_names</span><span class="p">]],</span>
    <span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="c1"># second dataset from &quot;The transcriptome dynamics of single cells during the cell cycle&quot;</span>
<span class="c1"># https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4226257</span>

<span class="n">adata_exon</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSM4226257_RAW/GSM4226257_out_gene_exon_tagged.dge_exonsds_046.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>
<span class="n">adata_intron</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">read_text</span><span class="p">(</span><span class="n">data_path</span> <span class="o">+</span> <span class="s1">&#39;GSM4226257_RAW/GSM4226257_out_gene_exon_tagged.dge_intronsds_046.txt&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">transpose</span><span class="p">()</span>

<span class="n">intersection</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">adata_exon</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata_intron</span><span class="o">.</span><span class="n">var_names</span><span class="p">))</span>
<span class="n">intersection_part</span> <span class="o">=</span> <span class="n">adata_exon</span><span class="p">[:,</span> <span class="n">intersection</span><span class="p">]</span>
<span class="n">intersection_part</span><span class="o">.</span><span class="n">X</span> <span class="o">+=</span> <span class="n">adata_intron</span><span class="p">[:,</span> <span class="n">intersection</span><span class="p">]</span><span class="o">.</span><span class="n">X</span>
<span class="n">adata1_3</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span>
        <span class="n">intersection_part</span><span class="p">,</span> 
        <span class="n">adata_exon</span><span class="p">[:,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">adata_exon</span><span class="o">.</span><span class="n">var_names</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata1_intron</span><span class="o">.</span><span class="n">var_names</span><span class="p">]],</span>
        <span class="n">adata_intron</span><span class="p">[:,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">adata_intron</span><span class="o">.</span><span class="n">var_names</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata1_exon</span><span class="o">.</span><span class="n">var_names</span><span class="p">]],</span>
    <span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">adata1_3</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>

<span class="n">tenx_adata</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">adata1_2</span><span class="p">,</span> <span class="n">adata2_2</span><span class="p">,</span> <span class="n">adata1_3</span><span class="p">],</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">,</span> <span class="n">fill_value</span><span class="o">=</span><span class="mf">0.</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Variable names are not unique. To make them unique, call `.var_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<section id="true-single-cell-proteomics-cell-cycle-analysis">
<h2><span class="section-number">1.1. </span>True Single-Cell Proteomics - Cell Cycle Analysis<a class="headerlink" href="#true-single-cell-proteomics-cell-cycle-analysis" title="Permalink to this headline">#</a></h2>
<section id="fig-4c-total-protein-signal-per-cell-grouped-by-cell-cycle-stage">
<h3><span class="section-number">1.1.1. </span>Fig. 4C - Total protein signal per cell grouped by cell cycle stage<a class="headerlink" href="#fig-4c-total-protein-signal-per-cell-grouped-by-cell-cycle-stage" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_raw</span> <span class="o">=</span> <span class="n">adata_prot_raw</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata_raw</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata_raw</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata_raw</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

<span class="n">adata_raw</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;G1&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G1_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G1-S&#39;</span> <span class="k">if</span> <span class="s1">&#39;_TB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G2_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2-M&#39;</span> <span class="k">if</span> <span class="s1">&#39;_NB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;other&#39;</span>
    <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">adata_raw</span><span class="o">.</span><span class="n">obs_names</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">size</span> <span class="o">=</span> <span class="p">{</span><span class="n">cond</span><span class="p">:</span> <span class="n">adata_raw</span><span class="p">[</span><span class="n">adata_raw</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">cond</span><span class="p">]</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
       <span class="k">for</span> <span class="n">cond</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;G1&#39;</span><span class="p">,</span> <span class="s1">&#39;G1-S&#39;</span><span class="p">,</span> <span class="s1">&#39;G2&#39;</span><span class="p">,</span> <span class="s1">&#39;G2-M&#39;</span><span class="p">]}</span>
<span class="n">x</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">([[</span><span class="n">cond</span><span class="p">]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">cells</span><span class="p">)</span> <span class="k">for</span> <span class="n">cond</span><span class="p">,</span> <span class="n">cells</span> <span class="ow">in</span> <span class="n">size</span><span class="o">.</span><span class="n">items</span><span class="p">()])</span>
<span class="n">y</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">([</span><span class="n">cells</span> <span class="k">for</span> <span class="n">cells</span> <span class="ow">in</span> <span class="n">size</span><span class="o">.</span><span class="n">values</span><span class="p">()])</span>
<span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Cell cycle stage&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Raw MS signal&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/seaborn/_decorators.py:36: FutureWarning: Pass the following variables as keyword args: x, y. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_11_1.png" src="../_images/TSP_cell_cycle_analysis_11_1.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">cond</span><span class="p">,</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">size</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">cond</span><span class="si">}</span><span class="s1">:</span><span class="se">\t</span><span class="s1"> n = </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="p">)</span><span class="si">}</span><span class="s1">,</span><span class="se">\t</span><span class="s1"> median = </span><span class="si">{</span><span class="n">np</span><span class="o">.</span><span class="n">median</span><span class="p">(</span><span class="n">d</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>G1:	 n = 84,	 median = 8477524.0
G1-S:	 n = 41,	 median = 12429287.0
G2:	 n = 52,	 median = 14910479.0
G2-M:	 n = 45,	 median = 9016382.0
</pre></div>
</div>
</div>
</div>
</section>
<section id="fig-4d-principal-component-analysis">
<h3><span class="section-number">1.1.2. </span>Fig. 4D - Principal Component Analysis<a class="headerlink" href="#fig-4d-principal-component-analysis" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">impute_downshifted_normal_global</span><span class="p">(</span>
        <span class="n">adata</span><span class="p">,</span>
        <span class="n">scale</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
        <span class="n">shift</span><span class="o">=</span><span class="mf">1.8</span><span class="p">,</span>
<span class="p">):</span>
    <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="mi">42</span><span class="p">)</span>
    <span class="n">mean</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
    <span class="n">std</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanstd</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
    <span class="n">draws</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">normal</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="n">mean</span> <span class="o">-</span> <span class="n">shift</span> <span class="o">*</span> <span class="n">std</span><span class="p">,</span> <span class="n">scale</span><span class="o">=</span><span class="n">scale</span> <span class="o">*</span> <span class="n">std</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)))</span>
    <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)]</span> <span class="o">=</span> <span class="n">draws</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">impute_downshifted_normal_global</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>

<span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;G1&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G1_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G1-S&#39;</span> <span class="k">if</span> <span class="s1">&#39;_TB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G2_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2-M&#39;</span> <span class="k">if</span> <span class="s1">&#39;_NB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;other&#39;</span>
    <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs_names</span>
<span class="p">]</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;other&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">,</span> <span class="n">annotate_var_explained</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1192: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if is_string_dtype(df[key]) and not is_categorical(df[key])
... storing &#39;cell cycle stage&#39; as categorical
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_16_1.png" src="../_images/TSP_cell_cycle_analysis_16_1.png" />
</div>
</div>
</section>
<section id="fig-4e-cell-cycle-stage-prediction">
<h3><span class="section-number">1.1.3. </span>Fig. 4E - Cell cycle stage prediction<a class="headerlink" href="#fig-4e-cell-cycle-stage-prediction" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.7</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>

<span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;G1&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G1_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G1-S&#39;</span> <span class="k">if</span> <span class="s1">&#39;_TB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G2_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2-M&#39;</span> <span class="k">if</span> <span class="s1">&#39;_NB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;other&#39;</span>
    <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs_names</span>
<span class="p">]</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;other&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cc_markers</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s2">&quot;CellCycleMarker.xlsx&quot;</span><span class="p">,</span> <span class="s2">&quot;Tami_Geiger&quot;</span><span class="p">)</span>

<span class="n">g1_list</span> <span class="o">=</span> <span class="n">cc_markers</span><span class="p">[</span><span class="s1">&#39;G1&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">g1_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">g</span> <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">g1_list</span> <span class="k">if</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;G1:</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">g1_list</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">s_list</span> <span class="o">=</span> <span class="n">cc_markers</span><span class="p">[</span><span class="s1">&#39;S&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">s_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">g</span> <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">s_list</span> <span class="k">if</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;S:</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">s_list</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">g2m_list</span> <span class="o">=</span> <span class="n">cc_markers</span><span class="p">[</span><span class="s1">&#39;G2M&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">g2m_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">g</span> <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">g2m_list</span> <span class="k">if</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;G2M:</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">g2m_list</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>G1:	 MARCKS, KRT1, HIST1H1E, KRT18, HNRNPA1, CHCHD3, CD44, NASP, TARDBP, PODXL, SUMO2, STMN1, TRIM28, SPTAN1
S:	 NOLC1, ATP2A2, CANX, TMX1, CKB, SLC25A3, SLC16A1, MT-CO2, SRPRB, CYB5R3, LETM1, ANP32B
G2M:	 TOP2A, HMGB1, EIF5B, TMSB10, EIF3D, ANP32A, RCC2, FASN, LUC7L2, AARS, KPNA2, CKAP5
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">score_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">g1_list</span><span class="p">,</span> <span class="n">score_name</span><span class="o">=</span><span class="s1">&#39;g1&#39;</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">score_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">s_list</span><span class="p">,</span> <span class="n">score_name</span><span class="o">=</span><span class="s1">&#39;s&#39;</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">score_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">g2m_list</span><span class="p">,</span> <span class="n">score_name</span><span class="o">=</span><span class="s1">&#39;g2m&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Trying to set attribute `.obs` of view, copying.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">protein_data_nb_tb</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;G2-M&#39;</span><span class="p">)</span> 
                                  <span class="o">|</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;G1-S&#39;</span><span class="p">)]</span>

<span class="n">score</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;G2-M&#39;</span><span class="p">:</span> <span class="n">protein_data_nb_tb</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;g2m&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">,</span>
    <span class="s1">&#39;S&#39;</span><span class="p">:</span> <span class="n">protein_data_nb_tb</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;s&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">,</span>
    <span class="s1">&#39;G1&#39;</span><span class="p">:</span> <span class="n">protein_data_nb_tb</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;g1&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">phase</span> <span class="ow">in</span> <span class="n">score</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">true_label</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">phase</span> <span class="ow">in</span> <span class="n">a</span> <span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">protein_data_nb_tb</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">])</span>
    <span class="n">scores</span> <span class="o">=</span> <span class="n">score</span><span class="p">[</span><span class="n">phase</span><span class="p">]</span>
    <span class="n">fp</span><span class="p">,</span> <span class="n">tp</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">sklearn</span><span class="o">.</span><span class="n">metrics</span><span class="o">.</span><span class="n">roc_curve</span><span class="p">(</span><span class="n">true_label</span><span class="p">,</span> <span class="n">scores</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">fp</span><span class="p">,</span> <span class="n">tp</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">phase</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">gca</span><span class="p">()</span><span class="o">.</span><span class="n">set_aspect</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;False-positive&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;True-positive&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;G2-M vs G1-S&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_21_0.png" src="../_images/TSP_cell_cycle_analysis_21_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">phase</span> <span class="ow">in</span> <span class="n">score</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">true_label</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">phase</span> <span class="ow">in</span> <span class="n">a</span> <span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">protein_data_nb_tb</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">])</span>
    <span class="n">scores</span> <span class="o">=</span> <span class="n">score</span><span class="p">[</span><span class="n">phase</span><span class="p">]</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;AUC using the </span><span class="si">{</span><span class="n">phase</span><span class="si">}</span><span class="s1"> score: </span><span class="si">{</span><span class="n">sklearn</span><span class="o">.</span><span class="n">metrics</span><span class="o">.</span><span class="n">roc_auc_score</span><span class="p">(</span><span class="n">true_label</span><span class="p">,</span><span class="w"> </span><span class="n">scores</span><span class="p">)</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AUC using the G2-M score: 0.94
AUC using the S score: 0.60
AUC using the G1 score: 0.74
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="true-single-cell-proteomics-comparison-with-transcriptomics-data">
<h2><span class="section-number">1.2. </span>True Single-Cell Proteomics - Comparison with Transcriptomics Data<a class="headerlink" href="#true-single-cell-proteomics-comparison-with-transcriptomics-data" title="Permalink to this headline">#</a></h2>
<section id="fig-5a-gene-expression-completeness-within-cells">
<h3><span class="section-number">1.2.1. </span>Fig. 5A - Gene expression completeness within cells<a class="headerlink" href="#fig-5a-gene-expression-completeness-within-cells" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMARTseq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">datasetname</span><span class="si">}</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s1">,</span><span class="se">\t</span><span class="s1"> mean completeness per cell: </span><span class="si">{</span><span class="n">np</span><span class="o">.</span><span class="n">median</span><span class="p">((</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">))</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Proteins	 (424, 2480),	 mean completeness per cell: 0.49
SMARTseq2	 (720, 24990),	 mean completeness per cell: 0.28
Drop-Seq	 (5022, 41161),	 mean completeness per cell: 0.07
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">perc_expr</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">dataset_name</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">perc_expr</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">((</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">))</span>
    <span class="n">dataset_name</span> <span class="o">+=</span> <span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">sns</span><span class="o">.</span><span class="n">violinplot</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">perc_expr</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">dataset_name</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Gene/Protein expression completeness per cell&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_27_0.png" src="../_images/TSP_cell_cycle_analysis_27_0.png" />
</div>
</div>
</section>
<section id="fig-5b-principal-component-analysis">
<h3><span class="section-number">1.2.2. </span>Fig. 5B - Principal Component Analysis<a class="headerlink" href="#fig-5b-principal-component-analysis" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
    
<span class="n">shared_genes</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">full_data</span><span class="p">[</span><span class="s1">&#39;Proteins&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>
<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">shared_genes</span> <span class="o">=</span> <span class="n">shared_genes</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>
    
<span class="n">intersection_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="n">datasetname</span><span class="p">:</span> <span class="n">adata</span><span class="p">[:,</span> <span class="nb">list</span><span class="p">(</span><span class="n">shared_genes</span><span class="p">)]</span>
    <span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">()</span>
<span class="p">}</span>
    
<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">intersection_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">target_sum</span><span class="o">=</span><span class="mf">1e6</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
    
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">intersection_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;dataset&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">datasetname</span>
<span class="n">joined</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">adata</span> <span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">intersection_data</span><span class="o">.</span><span class="n">values</span><span class="p">()])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/scanpy/preprocessing/_normalization.py:155: UserWarning: Revieved a view of an AnnData. Making a copy.
  view_to_actual(adata)
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">intersection_data</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>{&#39;Proteins&#39;: AnnData object with n_obs Ã— n_vars = 424 Ã— 1672
     obs: &#39;n_genes&#39;, &#39;dataset&#39;
     var: &#39;n_cells&#39;,
 &#39;SMART-Seq2&#39;: AnnData object with n_obs Ã— n_vars = 720 Ã— 1672
     obs: &#39;n_genes&#39;, &#39;dataset&#39;
     var: &#39;n_cells&#39;,
 &#39;Drop-Seq&#39;: AnnData object with n_obs Ã— n_vars = 5022 Ã— 1672
     obs: &#39;n_genes&#39;, &#39;dataset&#39;
     var: &#39;n_cells&#39;}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">joined</span><span class="p">)</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">joined</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;dataset&#39;</span><span class="p">,</span> <span class="n">show</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">annotate_var_explained</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">invert_xaxis</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1192: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if is_string_dtype(df[key]) and not is_categorical(df[key])
... storing &#39;dataset&#39; as categorical
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_31_1.png" src="../_images/TSP_cell_cycle_analysis_31_1.png" />
</div>
</div>
</section>
<section id="fig-5c-pearson-correlation-analysis">
<h3><span class="section-number">1.2.3. </span>Fig. 5C - Pearson Correlation Analysis<a class="headerlink" href="#fig-5c-pearson-correlation-analysis" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

<span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">name</span> <span class="o">!=</span> <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span>
        <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">target_sum</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nansum</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)))</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">shuffle</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>

<span class="n">joined</span> <span class="o">=</span> <span class="n">anndata</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">())</span> <span class="c1"># performs inner join</span>
<span class="n">dataset_names</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">([[</span><span class="n">name</span><span class="p">]</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">()])</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;number of shared genes:&#39;</span><span class="p">,</span> <span class="n">joined</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="n">c</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">joined</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">T</span><span class="p">)</span><span class="o">.</span><span class="n">corr</span><span class="p">())</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>number of shared genes: 1672
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">top</span> <span class="o">=</span> <span class="n">cm</span><span class="o">.</span><span class="n">get_cmap</span><span class="p">(</span><span class="s1">&#39;Blues_r&#39;</span><span class="p">,</span> <span class="mi">128</span><span class="p">)</span>
<span class="n">bottom</span> <span class="o">=</span> <span class="n">cm</span><span class="o">.</span><span class="n">get_cmap</span><span class="p">(</span><span class="s1">&#39;Oranges&#39;</span><span class="p">,</span> <span class="mi">128</span><span class="p">)</span>
<span class="n">newcolors</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">vstack</span><span class="p">((</span><span class="n">top</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">128</span><span class="p">)),</span>
                       <span class="n">bottom</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">128</span><span class="p">))))</span>
<span class="n">newcmp</span> <span class="o">=</span> <span class="n">ListedColormap</span><span class="p">(</span><span class="n">newcolors</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;OrangeBlue&#39;</span><span class="p">)</span>

<span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span><span class="mi">5</span><span class="p">),</span> <span class="n">gridspec_kw</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">wspace</span><span class="o">=</span><span class="mf">.03</span><span class="p">,</span> <span class="n">hspace</span><span class="o">=</span><span class="mf">.03</span><span class="p">))</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">dataset_1</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">full_data</span><span class="o">.</span><span class="n">keys</span><span class="p">()):</span>
    <span class="k">for</span> <span class="n">j</span><span class="p">,</span> <span class="n">dataset_2</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">full_data</span><span class="o">.</span><span class="n">keys</span><span class="p">()):</span>
        <span class="n">subdata</span> <span class="o">=</span> <span class="n">c</span><span class="p">[</span><span class="n">dataset_names</span> <span class="o">==</span> <span class="n">dataset_1</span><span class="p">][:,</span> <span class="n">dataset_names</span> <span class="o">==</span> <span class="n">dataset_2</span><span class="p">]</span>
        <span class="n">color_obj</span> <span class="o">=</span> <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">subdata</span><span class="p">,</span> <span class="n">vmin</span><span class="o">=-</span><span class="mi">1</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;auto&#39;</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">newcmp</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
        <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">dataset_2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">11</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">j</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="n">j</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">dataset_1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">11</span><span class="p">)</span>

<span class="n">cbar_ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_axes</span><span class="p">([</span><span class="mf">.92</span><span class="p">,</span> <span class="mf">0.15</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">,</span> <span class="mf">.7</span><span class="p">])</span>
<span class="n">plt</span><span class="o">.</span><span class="n">colorbar</span><span class="p">(</span><span class="n">color_obj</span><span class="p">,</span> <span class="n">cax</span><span class="o">=</span><span class="n">cbar_ax</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_35_0.png" src="../_images/TSP_cell_cycle_analysis_35_0.png" />
</div>
</div>
</section>
<section id="fig-5d-variance-analysis">
<h3><span class="section-number">1.2.4. </span>Fig. 5D - Variance analysis<a class="headerlink" href="#fig-5d-variance-analysis" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dir_core_proteome</span> <span class="o">=</span> <span class="s1">&#39;CoreProteome.txt&#39;</span>
<span class="n">core_proteins</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">dir_core_proteome</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)[</span><span class="s1">&#39;Genes&#39;</span><span class="p">])</span>

<span class="n">color</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Core proteome genes&#39;</span><span class="p">:</span> <span class="n">newcmp</span><span class="p">(</span><span class="mf">.85</span><span class="p">),</span>
    <span class="s1">&#39;Other proteins&#39;</span><span class="p">:</span> <span class="n">newcmp</span><span class="p">(</span><span class="mf">0.15</span><span class="p">)</span>
<span class="p">}</span>
<span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;Core proteome genes&#39;</span> <span class="k">if</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">core_proteins</span> 
    <span class="k">else</span> <span class="s1">&#39;Other proteins&#39;</span> 
    <span class="k">for</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span>
<span class="p">]</span>
<span class="n">mean</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">coeff_of_var</span> <span class="o">=</span> <span class="n">scipy</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">variation</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">nan_policy</span><span class="o">=</span><span class="s1">&#39;omit&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">data</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">JointGrid</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">mean</span><span class="p">),</span>
    <span class="n">y</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">coeff_of_var</span><span class="p">),</span>
    <span class="n">hue</span><span class="o">=</span><span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">],</span>
    <span class="n">palette</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">plot_joint</span><span class="p">(</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">,</span> 
    <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> 
    <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
    <span class="n">linewidth</span><span class="o">=</span><span class="mf">.1</span>
<span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">plot_marginals</span><span class="p">(</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">,</span> 
    <span class="n">kde</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> 
    <span class="n">stat</span><span class="o">=</span><span class="s1">&#39;density&#39;</span><span class="p">,</span> 
    <span class="n">common_norm</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
    <span class="n">binwidth</span><span class="o">=</span><span class="mf">.2</span><span class="p">,</span>
    <span class="n">alpha</span><span class="o">=</span><span class="mf">.8</span>
<span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;log 2 mean&#39;</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;log 2 coefficient of variation&#39;</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">4</span><span class="p">,</span> <span class="mi">18</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="o">-</span><span class="mi">6</span><span class="p">,</span> <span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">3</span><span class="p">,</span> <span class="o">-</span><span class="mi">5</span><span class="p">],</span> <span class="s1">&#39;--&#39;</span><span class="p">,</span><span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Poisson distribution&#39;</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="p">[])</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="n">datasetname</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_38_0.png" src="../_images/TSP_cell_cycle_analysis_38_0.png" />
</div>
</div>
</section>
<section id="fig-5e-variance-comparison">
<h3><span class="section-number">1.2.5. </span>Fig. 5E - Variance comparison<a class="headerlink" href="#fig-5e-variance-comparison" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

<span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">name</span> <span class="o">!=</span> <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span>
        <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">target_sum</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nansum</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">hue</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">dataset_name</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Core proteome genes&#39;</span> <span class="k">if</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">core_proteins</span> <span class="k">else</span> <span class="s1">&#39;Other proteins&#39;</span> <span class="k">for</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
    <span class="n">mean</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">coeff_of_var</span> <span class="o">=</span> <span class="n">scipy</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">variation</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">nan_policy</span><span class="o">=</span><span class="s1">&#39;omit&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">data</span>
    <span class="n">cov</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">coeff_of_var</span><span class="p">)</span>
    <span class="n">hue</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">])</span>
    <span class="n">dataset_name</span> <span class="o">+=</span> <span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>

<span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">dataset_name</span><span class="p">,</span> 
    <span class="n">y</span><span class="o">=</span><span class="n">cov</span><span class="p">,</span> 
    <span class="n">hue</span><span class="o">=</span><span class="n">hue</span><span class="p">,</span>
    <span class="n">palette</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
    <span class="n">showfliers</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;coefficient of variation&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylim</span><span class="p">(</span><span class="n">top</span><span class="o">=</span><span class="mi">6</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_41_0.png" src="../_images/TSP_cell_cycle_analysis_41_0.png" />
</div>
</div>
</section>
</section>
<section id="supplementary-analyses">
<h2><span class="section-number">1.3. </span>Supplementary Analyses<a class="headerlink" href="#supplementary-analyses" title="Permalink to this headline">#</a></h2>
<section id="effect-of-local-regression-normalisation">
<h3><span class="section-number">1.3.1. </span>Effect of Local Regression Normalisation<a class="headerlink" href="#effect-of-local-regression-normalisation" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata_normalised</span> <span class="o">=</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata_normalised</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata_normalised</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata_normalised</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata_normalised</span><span class="p">)</span>
<span class="n">adata_normalised</span><span class="o">.</span><span class="n">X</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">adata_normalised</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>

<span class="n">adata_raw</span> <span class="o">=</span> <span class="n">adata_prot_raw</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata_raw</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata_raw</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata_raw</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata_raw</span><span class="p">)</span>
<span class="n">adata_raw</span><span class="o">.</span><span class="n">X</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">adata_raw</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">x</span> <span class="o">=</span> <span class="n">adata_raw</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">y</span> <span class="o">=</span> <span class="p">(</span><span class="n">adata_raw</span><span class="o">.</span><span class="n">X</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;Raw MS-Signal per cell&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Identified proteins per cell&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">x</span> <span class="o">=</span> <span class="n">adata_normalised</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">y</span> <span class="o">=</span> <span class="p">(</span><span class="n">adata_normalised</span><span class="o">.</span><span class="n">X</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;Normalized MS-Signal per cell&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Identified proteins per cell&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_45_0.png" src="../_images/TSP_cell_cycle_analysis_45_0.png" />
<img alt="../_images/TSP_cell_cycle_analysis_45_1.png" src="../_images/TSP_cell_cycle_analysis_45_1.png" />
</div>
</div>
</section>
<section id="pca-including-untreated-cells">
<h3><span class="section-number">1.3.2. </span>PCA including untreated cells<a class="headerlink" href="#pca-including-untreated-cells" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">impute_downshifted_normal_global</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>

<span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;condition&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;G1&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G1_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G1-S&#39;</span> <span class="k">if</span> <span class="s1">&#39;_TB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G2_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2-M&#39;</span> <span class="k">if</span> <span class="s1">&#39;_NB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;UB_Passage01&#39;</span> <span class="k">if</span> <span class="s1">&#39;UB&#39;</span> <span class="ow">in</span> <span class="n">name</span> <span class="ow">and</span> <span class="s1">&#39;20201010&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;UB_Passage02_Plate01&#39;</span> <span class="k">if</span> <span class="s1">&#39;UB&#39;</span> <span class="ow">in</span> <span class="n">name</span> <span class="ow">and</span> <span class="nb">int</span><span class="p">(</span><span class="n">name</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;_&#39;</span><span class="p">)[</span><span class="mi">16</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;.&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span> <span class="o">&lt;</span> <span class="mi">5050</span>
    <span class="k">else</span> <span class="s1">&#39;UB_Passage02_Plate02&#39;</span>
    <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs_names</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;condition&#39;</span><span class="p">,</span> <span class="n">annotate_var_explained</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1192: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if is_string_dtype(df[key]) and not is_categorical(df[key])
... storing &#39;condition&#39; as categorical
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_48_1.png" src="../_images/TSP_cell_cycle_analysis_48_1.png" />
</div>
</div>
</section>
<section id="cell-cycle-stage-prediction">
<h3><span class="section-number">1.3.3. </span>Cell cycle stage prediction<a class="headerlink" href="#cell-cycle-stage-prediction" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.7</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>

<span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;G1&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G1_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G1-S&#39;</span> <span class="k">if</span> <span class="s1">&#39;_TB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2&#39;</span> <span class="k">if</span> <span class="s1">&#39;_G2_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;G2-M&#39;</span> <span class="k">if</span> <span class="s1">&#39;_NB_&#39;</span> <span class="ow">in</span> <span class="n">name</span>
    <span class="k">else</span> <span class="s1">&#39;other&#39;</span>
    <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs_names</span>
<span class="p">]</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;other&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cc_markers</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s2">&quot;CellCycleMarker.xlsx&quot;</span><span class="p">,</span> <span class="s2">&quot;Tami_Geiger&quot;</span><span class="p">)</span>

<span class="n">g1_list</span> <span class="o">=</span> <span class="n">cc_markers</span><span class="p">[</span><span class="s1">&#39;G1&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">g1_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">g</span> <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">g1_list</span> <span class="k">if</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;G1:</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">g1_list</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">s_list</span> <span class="o">=</span> <span class="n">cc_markers</span><span class="p">[</span><span class="s1">&#39;S&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">s_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">g</span> <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">s_list</span> <span class="k">if</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;S:</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">s_list</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">g2m_list</span> <span class="o">=</span> <span class="n">cc_markers</span><span class="p">[</span><span class="s1">&#39;G2M&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">g2m_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">g</span> <span class="k">for</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">g2m_list</span> <span class="k">if</span> <span class="n">g</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;G2M:</span><span class="se">\t</span><span class="s1"> </span><span class="si">{</span><span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">g2m_list</span><span class="p">)</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>G1:	 MARCKS, KRT1, HIST1H1E, KRT18, HNRNPA1, CHCHD3, CD44, NASP, TARDBP, PODXL, SUMO2, STMN1, TRIM28, SPTAN1
S:	 NOLC1, ATP2A2, CANX, TMX1, CKB, SLC25A3, SLC16A1, MT-CO2, SRPRB, CYB5R3, LETM1, ANP32B
G2M:	 TOP2A, HMGB1, EIF5B, TMSB10, EIF3D, ANP32A, RCC2, FASN, LUC7L2, AARS, KPNA2, CKAP5
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">score_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">g1_list</span><span class="p">,</span> <span class="n">score_name</span><span class="o">=</span><span class="s1">&#39;g1&#39;</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">score_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">s_list</span><span class="p">,</span> <span class="n">score_name</span><span class="o">=</span><span class="s1">&#39;s&#39;</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">score_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">g2m_list</span><span class="p">,</span> <span class="n">score_name</span><span class="o">=</span><span class="s1">&#39;g2m&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Trying to set attribute `.obs` of view, copying.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">pair</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;G2&#39;</span><span class="p">,</span> <span class="s1">&#39;G1&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;G2-M&#39;</span><span class="p">,</span> <span class="s1">&#39;G1&#39;</span><span class="p">)]:</span>
    <span class="n">data_comparison</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span> 
                                      <span class="o">|</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">])]</span>

    <span class="n">score</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="n">score</span><span class="p">[</span><span class="s1">&#39;G2-M&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data_comparison</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;g2m&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">score</span><span class="p">[</span><span class="s1">&#39;S&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data_comparison</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;s&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    <span class="n">score</span><span class="p">[</span><span class="s1">&#39;G1&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data_comparison</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;g1&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
    
    <span class="k">for</span> <span class="n">phase</span> <span class="ow">in</span> <span class="n">score</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
        <span class="n">true_label</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">phase</span> <span class="ow">in</span> <span class="n">a</span> <span class="ow">or</span> <span class="p">(</span><span class="n">phase</span> <span class="o">==</span> <span class="s1">&#39;S&#39;</span> <span class="ow">and</span> <span class="n">a</span> <span class="o">==</span> <span class="s1">&#39;G1&#39;</span><span class="p">)</span> <span class="ow">or</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">phase</span> <span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">data_comparison</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;cell cycle stage&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">])</span>
        <span class="n">scores</span> <span class="o">=</span> <span class="n">score</span><span class="p">[</span><span class="n">phase</span><span class="p">]</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;AUC using the </span><span class="si">{</span><span class="n">phase</span><span class="si">}</span><span class="s1"> score: </span><span class="si">{</span><span class="n">sklearn</span><span class="o">.</span><span class="n">metrics</span><span class="o">.</span><span class="n">roc_auc_score</span><span class="p">(</span><span class="n">true_label</span><span class="p">,</span><span class="w"> </span><span class="n">scores</span><span class="p">)</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fp</span><span class="p">,</span> <span class="n">tp</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">sklearn</span><span class="o">.</span><span class="n">metrics</span><span class="o">.</span><span class="n">roc_curve</span><span class="p">(</span><span class="n">true_label</span><span class="p">,</span> <span class="n">scores</span><span class="p">)</span>
        <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">fp</span><span class="p">,</span> <span class="n">tp</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">phase</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">gca</span><span class="p">()</span><span class="o">.</span><span class="n">set_aspect</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;False-positive&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;True-positive&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> vs </span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AUC using the G2-M score: 0.46
AUC using the S score: 0.35
AUC using the G1 score: 0.67
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_53_1.png" src="../_images/TSP_cell_cycle_analysis_53_1.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AUC using the G2-M score: 0.66
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>AUC using the S score: 0.52
AUC using the G1 score: 0.84
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_53_5.png" src="../_images/TSP_cell_cycle_analysis_53_5.png" />
</div>
</div>
</section>
<section id="data-completeness">
<h3><span class="section-number">1.3.4. </span>Data completeness<a class="headerlink" href="#data-completeness" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">shared_genes</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">full_data</span><span class="p">[</span><span class="s1">&#39;Proteins&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>
<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">shared_genes</span> <span class="o">=</span> <span class="n">shared_genes</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>
    
<span class="n">intersection_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="n">datasetname</span><span class="p">:</span> <span class="n">adata</span><span class="p">[:,</span> <span class="nb">list</span><span class="p">(</span><span class="n">shared_genes</span><span class="p">)]</span>
    <span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">()</span>
<span class="p">}</span>

<span class="n">perc_expr</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">dataset_name</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">intersection_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">datasetname</span><span class="si">}</span><span class="s1"> </span><span class="si">{</span><span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">perc_expr</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">((</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">))</span>
    <span class="n">dataset_name</span> <span class="o">+=</span> <span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">sns</span><span class="o">.</span><span class="n">violinplot</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">perc_expr</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">dataset_name</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Gene/Protein expression completeness per cell&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Proteins (424, 2447)
SMART-Seq2 (720, 2447)
Drop-Seq (5022, 2447)
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_56_1.png" src="../_images/TSP_cell_cycle_analysis_56_1.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">completeness</span> <span class="o">=</span> <span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span> <span class="o">/</span> <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">sorted_completeness</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">completeness</span><span class="p">)[::</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">sorted_completeness</span><span class="p">))</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> <span class="n">sorted_completeness</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Gene/Protein completeness&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;Ranked protein (by completeness)&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xscale</span><span class="p">(</span><span class="s2">&quot;log&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_57_0.png" src="../_images/TSP_cell_cycle_analysis_57_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">y</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="n">x</span><span class="p">),</span> <span class="n">y</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">datasetname</span><span class="p">)</span>
    
<span class="c1"># Pois(0) = exp(-mean) -&gt; prob for k zeros in n cells = Binom(k|p, n), with expectation p*n -&gt; divide by n</span>
<span class="n">x2</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1000000</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="n">x2</span><span class="p">),</span> <span class="mi">1</span><span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">exp</span><span class="p">(</span><span class="o">-</span><span class="n">x2</span><span class="p">),</span> <span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;expected poisson dropout&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;log mean protein abundance&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Completeness accross cells&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">markerscale</span><span class="o">=</span><span class="mf">5.</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;ipython-input-43-a69cb3e7f1f8&gt;:4: RuntimeWarning: divide by zero encountered in log
  plt.scatter(np.log(x), y, s=1, label=datasetname)
&lt;ipython-input-43-a69cb3e7f1f8&gt;:8: RuntimeWarning: divide by zero encountered in log
  plt.plot(np.log(x2), 1-np.exp(-x2), c=&#39;black&#39;, label=&#39;expected poisson dropout&#39;)
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_58_1.png" src="../_images/TSP_cell_cycle_analysis_58_1.png" />
</div>
</div>
</section>
<section id="variance-analyses">
<h3><span class="section-number">1.3.5. </span>Variance analyses<a class="headerlink" href="#variance-analyses" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

<span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">name</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nansum</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)))</span>
    <span class="k">if</span> <span class="n">name</span> <span class="o">!=</span> <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span>
        <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">target_sum</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nansum</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Proteins 9631877.0
SMART-Seq2 231281.56
Drop-Seq 7808.1196
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mean</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">coeff_of_var</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">name</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">mean</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">))</span>
    <span class="n">coeff_of_var</span> <span class="o">+=</span>  <span class="nb">list</span><span class="p">(</span><span class="n">scipy</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">variation</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">nan_policy</span><span class="o">=</span><span class="s1">&#39;omit&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>
    <span class="n">name</span> <span class="o">+=</span> <span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">))</span>
    
<span class="n">color</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="s1">&#39;red&#39;</span><span class="p">,</span>
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="s1">&#39;blue&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="s1">&#39;yellow&#39;</span><span class="p">,</span>
<span class="p">}</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">JointGrid</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">mean</span><span class="p">),</span>
    <span class="n">y</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">coeff_of_var</span><span class="p">),</span>
    <span class="n">hue</span><span class="o">=</span><span class="n">name</span><span class="p">,</span>
    <span class="n">palette</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">plot_joint</span><span class="p">(</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">,</span> 
    <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> 
    <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
    <span class="n">linewidth</span><span class="o">=</span><span class="mf">.1</span>
<span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">plot_marginals</span><span class="p">(</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">,</span> 
    <span class="n">kde</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> 
    <span class="n">stat</span><span class="o">=</span><span class="s1">&#39;density&#39;</span><span class="p">,</span> 
    <span class="n">common_norm</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
    <span class="n">binwidth</span><span class="o">=</span><span class="mf">.2</span><span class="p">,</span>
    <span class="n">alpha</span><span class="o">=</span><span class="mf">.8</span>
<span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;log 2 mean&#39;</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;log 2 coefficient of variation&#39;</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">4</span><span class="p">,</span> <span class="mi">19</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mi">4</span><span class="p">,</span> <span class="mf">4.2</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="o">-</span><span class="mi">6</span><span class="p">,</span> <span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">3</span><span class="p">,</span> <span class="o">-</span><span class="mi">5</span><span class="p">],</span> <span class="s1">&#39;--&#39;</span><span class="p">,</span><span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Poisson distribution&#39;</span><span class="p">)</span>
<span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="p">[])</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_61_0.png" src="../_images/TSP_cell_cycle_analysis_61_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dir_core_proteome</span> <span class="o">=</span> <span class="s1">&#39;CoreProteome.txt&#39;</span>
<span class="n">core_proteins</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">dir_core_proteome</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)[</span><span class="s1">&#39;Genes&#39;</span><span class="p">])</span>

<span class="n">cov</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">hue</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">dataset_name</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">color</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Core proteome genes&#39;</span><span class="p">:</span> <span class="n">newcmp</span><span class="p">(</span><span class="mf">.85</span><span class="p">),</span>
    <span class="s1">&#39;Other proteins&#39;</span><span class="p">:</span> <span class="n">newcmp</span><span class="p">(</span><span class="mf">0.15</span><span class="p">)</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>    
    <span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Core proteome genes&#39;</span> <span class="k">if</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">core_proteins</span> <span class="k">else</span> <span class="s1">&#39;Other proteins&#39;</span> <span class="k">for</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">]</span>
    <span class="n">mean</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">coeff_of_var</span> <span class="o">=</span> <span class="n">scipy</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">variation</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">nan_policy</span><span class="o">=</span><span class="s1">&#39;omit&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">data</span>
    <span class="n">cov</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">coeff_of_var</span><span class="p">)</span>
    <span class="n">hue</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">])</span>
    <span class="n">g</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">JointGrid</span><span class="p">(</span>
        <span class="n">x</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">mean</span><span class="p">),</span>
        <span class="n">y</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">coeff_of_var</span><span class="p">),</span>
        <span class="n">hue</span><span class="o">=</span><span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;core/noncore&#39;</span><span class="p">],</span>
        <span class="n">palette</span><span class="o">=</span><span class="n">color</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">plot_joint</span><span class="p">(</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">,</span> 
        <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> 
        <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
        <span class="n">linewidth</span><span class="o">=</span><span class="mf">.1</span>
    <span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">plot_marginals</span><span class="p">(</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">,</span> 
        <span class="n">kde</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> 
        <span class="n">stat</span><span class="o">=</span><span class="s1">&#39;density&#39;</span><span class="p">,</span> 
        <span class="n">common_norm</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="n">binwidth</span><span class="o">=</span><span class="mf">.2</span><span class="p">,</span>
        <span class="n">alpha</span><span class="o">=</span><span class="mf">.8</span>
    <span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;log 2 mean&#39;</span><span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;log 2 coefficient of variation&#39;</span><span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">4</span><span class="p">,</span> <span class="mi">19</span><span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mi">4</span><span class="p">,</span> <span class="mf">4.2</span><span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="o">-</span><span class="mi">6</span><span class="p">,</span> <span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">3</span><span class="p">,</span> <span class="o">-</span><span class="mi">5</span><span class="p">],</span> <span class="s1">&#39;--&#39;</span><span class="p">,</span><span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Poisson distribution&#39;</span><span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_joint</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="p">[])</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="n">datasetname</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_62_0.png" src="../_images/TSP_cell_cycle_analysis_62_0.png" />
<img alt="../_images/TSP_cell_cycle_analysis_62_1.png" src="../_images/TSP_cell_cycle_analysis_62_1.png" />
<img alt="../_images/TSP_cell_cycle_analysis_62_2.png" src="../_images/TSP_cell_cycle_analysis_62_2.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">inter</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">full_data</span><span class="p">[</span><span class="s1">&#39;Proteins&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>
<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">inter</span> <span class="o">=</span> <span class="n">inter</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>

<span class="n">intersection_data</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">intersection_data</span><span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[:,</span> <span class="nb">list</span><span class="p">(</span><span class="n">inter</span><span class="p">)]</span>

<span class="n">cvs</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">intersection_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">cvs</span><span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">=</span> <span class="n">scipy</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">variation</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">nan_policy</span><span class="o">=</span><span class="s1">&#39;omit&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">data</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">x</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">y</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">pair</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;Proteins&#39;</span><span class="p">,</span> <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;Proteins&#39;</span><span class="p">,</span> <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">,</span> <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">)]:</span>
    <span class="n">r</span> <span class="o">=</span> <span class="n">scipy</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">cvs</span><span class="p">[</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]]),</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">cvs</span><span class="p">[</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]]))[</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">x</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> vs </span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">y</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">r</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> 
    <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Pearson</span><span class="se">\&#39;</span><span class="s1">s Correlation between log2 cv&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">1.01</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_64_0.png" src="../_images/TSP_cell_cycle_analysis_64_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">pair</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">,</span> <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;Proteins&#39;</span><span class="p">,</span> <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;Proteins&#39;</span><span class="p">,</span> <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">)]:</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span><span class="mi">4</span><span class="p">))</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">cvs</span><span class="p">[</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]]),</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">cvs</span><span class="p">[</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]]),</span> <span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;log2 cv </span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;log2 cv </span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_65_0.png" src="../_images/TSP_cell_cycle_analysis_65_0.png" />
<img alt="../_images/TSP_cell_cycle_analysis_65_1.png" src="../_images/TSP_cell_cycle_analysis_65_1.png" />
<img alt="../_images/TSP_cell_cycle_analysis_65_2.png" src="../_images/TSP_cell_cycle_analysis_65_2.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">pair</span> <span class="ow">in</span> <span class="p">[(</span><span class="s1">&#39;Proteins&#39;</span><span class="p">,</span> <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">),</span> <span class="p">(</span><span class="s1">&#39;Proteins&#39;</span><span class="p">,</span> <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">)]:</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span><span class="mi">4</span><span class="p">))</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">cvs</span><span class="p">[</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]],</span> <span class="n">cvs</span><span class="p">[</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]],</span> <span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;CV(</span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">)&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;CV(</span><span class="si">{</span><span class="n">pair</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">)&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="o">-</span><span class="mi">1</span><span class="p">,</span> <span class="mi">5</span><span class="p">],</span> <span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">,</span> <span class="mi">5</span><span class="p">],</span> <span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s1">&#39;equal&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">gca</span><span class="p">()</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">gca</span><span class="p">()</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mf">.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_66_0.png" src="../_images/TSP_cell_cycle_analysis_66_0.png" />
<img alt="../_images/TSP_cell_cycle_analysis_66_1.png" src="../_images/TSP_cell_cycle_analysis_66_1.png" />
</div>
</div>
</section>
<section id="correlation-analyses">
<h3><span class="section-number">1.3.6. </span>Correlation analyses<a class="headerlink" href="#correlation-analyses" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

<span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">name</span> <span class="o">!=</span> <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span>
        <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">target_sum</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nansum</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)))</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
    
<span class="n">shared_genes</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">full_data</span><span class="p">[</span><span class="s1">&#39;Proteins&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>
<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">shared_genes</span> <span class="o">=</span> <span class="n">shared_genes</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">var_names</span><span class="p">)</span>

<span class="n">intersection_data</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">intersection_data</span><span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[:,</span> <span class="nb">list</span><span class="p">(</span><span class="n">shared_genes</span><span class="p">)]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cell1</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">full_data</span><span class="p">[</span><span class="s1">&#39;Proteins&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">X</span><span class="p">[</span><span class="mi">14</span><span class="p">,</span> <span class="p">:]</span><span class="o">.</span><span class="n">copy</span><span class="p">())</span>
<span class="n">cell2</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan_to_num</span><span class="p">(</span><span class="n">full_data</span><span class="p">[</span><span class="s1">&#39;Proteins&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">X</span><span class="p">[</span><span class="mi">15</span><span class="p">,</span> <span class="p">:]</span><span class="o">.</span><span class="n">copy</span><span class="p">())</span>

<span class="n">grid</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">jointplot</span><span class="p">(</span><span class="n">cell1</span><span class="p">,</span> <span class="n">cell2</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">marginal_kws</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;bins&#39;</span><span class="p">:</span> <span class="mi">20</span><span class="p">})</span>
<span class="n">grid</span><span class="o">.</span><span class="n">set_axis_labels</span><span class="p">(</span><span class="s1">&#39;log intensity cell #1&#39;</span><span class="p">,</span> <span class="s1">&#39;log intensity cell #2&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/seaborn/_decorators.py:36: FutureWarning: Pass the following variables as keyword args: x, y. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(
</pre></div>
</div>
<img alt="../_images/TSP_cell_cycle_analysis_69_1.png" src="../_images/TSP_cell_cycle_analysis_69_1.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">r_values</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">data_type</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">r</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">corr</span><span class="p">())</span>
    <span class="n">np</span><span class="o">.</span><span class="n">fill_diagonal</span><span class="p">(</span><span class="n">r</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">)</span>
    <span class="n">r_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">val</span> <span class="k">for</span> <span class="n">val</span> <span class="ow">in</span> <span class="n">r</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">val</span><span class="p">)]</span>
    <span class="n">r_values</span> <span class="o">+=</span> <span class="n">r_vals</span>
    <span class="n">data_type</span> <span class="o">+=</span> <span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">r_vals</span><span class="p">)</span>

<span class="n">r_vals</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">r_values</span><span class="p">)</span>
<span class="k">for</span> <span class="n">datasetname</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">corrs</span> <span class="o">=</span> <span class="n">r_vals</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">data_type</span><span class="p">)</span> <span class="o">==</span> <span class="n">datasetname</span><span class="p">]</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">data_type</span><span class="p">,</span>
    <span class="n">y</span><span class="o">=</span><span class="n">r_values</span><span class="p">,</span>
    <span class="n">width</span><span class="o">=</span><span class="mf">0.6</span><span class="p">,</span>
    <span class="n">showfliers</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="p">)</span>        
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Pearson correlation between two cells&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">1.01</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;All genes&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_70_0.png" src="../_images/TSP_cell_cycle_analysis_70_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">r_values</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">data_type</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">datasetname</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">intersection_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">r</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">corr</span><span class="p">())</span>
    <span class="n">np</span><span class="o">.</span><span class="n">fill_diagonal</span><span class="p">(</span><span class="n">r</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">)</span>
    <span class="n">r_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">val</span> <span class="k">for</span> <span class="n">val</span> <span class="ow">in</span> <span class="n">r</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">val</span><span class="p">)]</span>
    <span class="n">r_values</span> <span class="o">+=</span> <span class="n">r_vals</span>
    <span class="n">data_type</span> <span class="o">+=</span> <span class="p">[</span><span class="n">datasetname</span><span class="p">]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">r_vals</span><span class="p">)</span>

<span class="n">r_vals</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">r_values</span><span class="p">)</span>
<span class="k">for</span> <span class="n">datasetname</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">corrs</span> <span class="o">=</span> <span class="n">r_vals</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">data_type</span><span class="p">)</span> <span class="o">==</span> <span class="n">datasetname</span><span class="p">]</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">boxplot</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">data_type</span><span class="p">,</span>
    <span class="n">y</span><span class="o">=</span><span class="n">r_values</span><span class="p">,</span>
    <span class="n">width</span><span class="o">=</span><span class="mf">0.6</span><span class="p">,</span>
    <span class="n">showfliers</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="p">)</span>        
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Pearson correlation between two cells&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">1.01</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Shared genes&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_71_0.png" src="../_images/TSP_cell_cycle_analysis_71_0.png" />
</div>
</div>
</section>
<section id="abundance-value-distributions">
<h3><span class="section-number">1.3.7. </span>Abundance value distributions<a class="headerlink" href="#abundance-value-distributions" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">vals</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">X</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>
    <span class="n">vals</span> <span class="o">=</span> <span class="n">vals</span><span class="p">[</span><span class="n">vals</span> <span class="o">!=</span> <span class="mi">0</span><span class="p">]</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">hist</span><span class="p">(</span><span class="n">vals</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;log_10 abundance&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Count&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/TSP_cell_cycle_analysis_74_0.png" src="../_images/TSP_cell_cycle_analysis_74_0.png" />
<img alt="../_images/TSP_cell_cycle_analysis_74_1.png" src="../_images/TSP_cell_cycle_analysis_74_1.png" />
<img alt="../_images/TSP_cell_cycle_analysis_74_2.png" src="../_images/TSP_cell_cycle_analysis_74_2.png" />
</div>
</div>
</section>
<section id="data-before-and-after-filtering">
<h3><span class="section-number">1.3.8. </span>Data before and after filtering<a class="headerlink" href="#data-before-and-after-filtering" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">full_data</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Proteins&#39;</span><span class="p">:</span> <span class="n">protein_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span> 
    <span class="s1">&#39;SMART-Seq2&#39;</span><span class="p">:</span> <span class="n">smartseq_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
    <span class="s1">&#39;Drop-Seq&#39;</span><span class="p">:</span> <span class="n">tenx_adata</span><span class="o">.</span><span class="n">copy</span><span class="p">(),</span>
<span class="p">}</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Before filtering:&#39;</span><span class="p">)</span>
<span class="n">full_data</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Before filtering:
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>{&#39;Proteins&#39;: AnnData object with n_obs Ã— n_vars = 434 Ã— 2480,
 &#39;SMART-Seq2&#39;: AnnData object with n_obs Ã— n_vars = 720 Ã— 24990,
 &#39;Drop-Seq&#39;: AnnData object with n_obs Ã— n_vars = 5665 Ã— 41161}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">adata</span> <span class="ow">in</span> <span class="n">full_data</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="mf">0.15</span> <span class="o">*</span> <span class="n">adata</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;After filtering:&#39;</span><span class="p">)</span>
<span class="n">full_data</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:1094: FutureWarning: is_categorical is deprecated and will be removed in a future version.  Use is_categorical_dtype instead
  if not is_categorical(df_full[k]):
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>After filtering:
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>{&#39;Proteins&#39;: AnnData object with n_obs Ã— n_vars = 424 Ã— 1867
     obs: &#39;n_genes&#39;
     var: &#39;n_cells&#39;,
 &#39;SMART-Seq2&#39;: AnnData object with n_obs Ã— n_vars = 720 Ã— 10557
     obs: &#39;n_genes&#39;
     var: &#39;n_cells&#39;,
 &#39;Drop-Seq&#39;: AnnData object with n_obs Ã— n_vars = 5022 Ã— 6701
     obs: &#39;n_genes&#39;
     var: &#39;n_cells&#39;}
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="../README.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Ultra-high Sensitivity Mass Spectrometry Single-Cell Proteome Analysis ðŸ”¬ ðŸ§¬ ðŸ’» ðŸ§ª</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="documentation.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">2. </span>Documentation</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Santiago Castro Dau<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>